src/port1.0/portbump.tcl

```diff
@@ -280,32 +282,71 @@ proc portbump::bump_main {args} {
                 return -code error "sed sed(1) failed"
             }
 
-            # Check Portfile overwrite permissios
-            if {[catch {file attributes $portfile -permissions u+w} error]} {
-                ui_debug $::errorInfo
-                ui_error "bump: $error"
-                file delete "$tmpfile"
-                return -code error "bump overwrite permissions failed"
-            }
+            if {[tbool ports_bump_patch]} {
+                # Patch mode
+                # Create Portfile.patch
+                # if {[catch {set patchfile [mkstemp "${tempdir}/Portfile.patch"]} error]} {
+                #     ui_debug $::errorInfo
+                #     ui_error "mkstemp: $error"
+                #     return -code error "mkstemp failed"
+                # } else {
+                #     # Extract the Tcl Channel number
+                #     set patchfd [lindex $patchfile 0]
+                #     # Set patchfile to only the file name
+                #     set patchfile [join [lrange $patchfile 1 end]]
+                # }
 
-            # Replace Portfile with Portfile.bump
-            if {[catch {file copy -force $tmpfile $portfile} error]} {
-                ui_debug $::errorInfo
-                ui_error "bump: $error"
-                file delete "$tmpfile"
-                return -code error "bump overwrite failed"
-            }
+                # Set Potfile.patch path
+                set patchfile "${tempdir}/Portfile.patch"
+
+                # Construct diff command
+                set diffcmd {}
+                lappend diffcmd $portutil::autoconf::diff_path -U 0
```

Comment:
Why `diff -U 0`? That is not really common... I would use just `diff -u`.

Furthermore, you could use something like `--label a/Portfile --label b/Portfile` to hide any temporary file names.

Reply:
I was thinking to shorten the lines. Yeah it might confuse people, I'll change it. Will add labels